# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/basetypes/TableCellAccessibleBase.h
# Commit: 6d0f4821ddcf
# Full Hash: 6d0f4821ddcfca1b59e4db3befc3ec9c12335891
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-03-23 09:49:32
# Regressor Bug: 1735970
# File Overlap Count: 1
# Description:
#   Bug 1735970 part 3: Add TableAccessibleBase and TableCellAccessibleBase. r=morgan
#   
#   These are needed to support RemoteAccessible tables.
#   Stuff specific to LocalAccessible is still in TableAccessible and TableCellAccessible, which now inherit from the new Base classes.
#   
# ==============================================================================

diff -r d82660fb5408 -r 6d0f4821ddcf accessible/basetypes/TableCellAccessibleBase.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/accessible/basetypes/TableCellAccessibleBase.h	Wed Mar 23 04:00:26 2022 +0000
@@ -0,0 +1,68 @@
+/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_a11y_TableCellAccessibleBase_h__
+#define mozilla_a11y_TableCellAccessibleBase_h__
+
+#include "nsTArray.h"
+#include <stdint.h>
+
+namespace mozilla {
+namespace a11y {
+
+class Accessible;
+class TableAccessibleBase;
+
+/**
+ * Abstract interface implemented by table cell accessibles.
+ */
+class TableCellAccessibleBase {
+ public:
+  /**
+   * Return the table this cell is in.
+   */
+  virtual TableAccessibleBase* Table() const = 0;
+
+  /**
+   * Return the column of the table this cell is in.
+   */
+  virtual uint32_t ColIdx() const = 0;
+
+  /**
+   * Return the row of the table this cell is in.
+   */
+  virtual uint32_t RowIdx() const = 0;
+
+  /**
+   * Return the column extent of this cell.
+   */
+  virtual uint32_t ColExtent() const { return 1; }
+
+  /**
+   * Return the row extent of this cell.
+   */
+  virtual uint32_t RowExtent() const { return 1; }
+
+  /**
+   * Return the column header cells for this cell.
+   */
+  virtual void ColHeaderCells(nsTArray<Accessible*>* aCells) = 0;
+
+  /**
+   * Return the row header cells for this cell.
+   */
+  virtual void RowHeaderCells(nsTArray<Accessible*>* aCells) = 0;
+
+  /**
+   * Returns true if this cell is selected.
+   */
+  virtual bool Selected() = 0;
+};
+
+}  // namespace a11y
+}  // namespace mozilla
+
+#endif  // mozilla_a11y_TableCellAccessibleBase_h__